### PR TITLE
chore: [app dir bootstrapping 7] update middleware

### PR DESCRIPTION
**Pull request by dmytro068**
## What does this PR do?

- Updates middleware matcher config to be executed on each page request
- Adds headers required by RootLayout (x-csp, x-url)

## Requirement/Documentation

- If there is a requirement document, please, share it here.
- If there is ab UI/UX design document, please, share it here.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] Chore (refactoring code, technical debt, workflow improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How should this be tested?

- Are there environment variables that should be set?
- What are the minimal test data to have?
- What is expected (happy path) to have (input and output)?
- Any other important info that could help to test that PR

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: apps/web/middleware.ts [hunk 1]
```diff
@@ -63,18 +79,8 @@ const routingForms = {
...
-  // https://github.com/vercel/next.js/discussions/42458
-  matcher: [
-    "/:path*/embed",
-    "/api/trpc/:path*",
-    "/login",
-    "/auth/login",
-    /**
-     * Paths required by routingForms.handle
-     */
-    "/apps/routing_forms/:path*",
```
Comment (zomars) [inline apps/web/middleware.ts:76]: Are these removals intended? tagging @hariombalhara who might have more context on this.

Comment (DmytroHryshyn) [reply apps/web/middleware.ts:76]: This change was required to ensure that middleware is executed on each page request. This is needed to set the `x-url` and `x-csp` headers that are used in RootLayout. With previous matcher settings we executed middleware only on some specific routes, which resulted in a bug where some app dir pages missed `x-url` header. 
We are open to  discuss alternative solutions. 
Also, we can add app dir pages paths to the matcher one by one as we migrate pages. So for current PR matcher will remain the same.

**Review by emrysal — APPROVED** [review]
Looks good to me, but ideally consider changing the req.headers map (when it is one) to a Headers object so you don't have to conditional as much.

**Comment by emrysal** [comment]
@hbjORbj Looks like the embeds tests are consistently failing, could you have look at that?

**Comment by hbjORbj** [comment]
> @hbjORbj Looks like the embeds tests are consistently failing, could you have look at that?

Thank you for catching that.

@DmytroHryshyn is taking care of this issue. Can you provide an ETA?

**Comment by DmytroHryshyn** [comment]
> > @hbjORbj Looks like the embeds tests are consistently failing, could you have look at that?
> 
> Thank you for catching that.
> 
> @DmytroHryshyn is taking care of this issue. Can you provide an ETA?

After rebasing against main, all CI steps are passing 
@emrysal

**Pull request merged**